modules/mauth-signer-akka-http/src/main/scala/com/mdsol/mauth/http/Implicits.scala

```diff
@@ -59,6 +63,17 @@ object Implicits {
   private def removeContentTypeFromHeaders(requestHeaders: Map[String, String]): Map[String, String] =
     requestHeaders.filterKeys(_ != headers.`Content-Type`.name)
 
+  private def getHttpEntity(contentTypeOptional: Option[String], entityBody: Array[Byte]) = {
+    contentTypeOptional match {
+      case Some(contentType) =>
+        ContentType.parse(contentType) match {
+          case Right(parsedContentType) => HttpEntity(parsedContentType, entityBody)
+          case _ => HttpEntity(DEFAULT_CONTENT_TYPE, entityBody)
+        }
+      case None => HttpEntity(DEFAULT_CONTENT_TYPE, entityBody)
+    }
+  }
```

Comment:
This seems overly complicated. Can we simply use the original headers? I'm playing around with this - will report back

Reply:
We need to remove from the original headers the content-type and add it to the HttpEntity

Reply:
You're right...Akka-http does not allow overwriting the headers 🙄